File: go/cmd/hello.go
package cmd

import (
	"fmt"
	"strings"

	"github.com/spf13/cobra"
)

var uppercase bool

var helloCmd = &cobra.Command{
	Use:   "hello <name>",
	Short: "Greet a user",
	Args:  cobra.ExactArgs(1),
	Run: func(cmd *cobra.Command, args []string) {
		greeting := fmt.Sprintf("Hello, %s!", args[0])
		if uppercase {
			greeting = strings.ToUpper(greeting)
		}
		fmt.Println(greeting)
	},
}

func init() {
	helloCmd.Flags().BoolVarP(&uppercase, "uppercase", "u", false, "print greeting in uppercase")
}

File: rust/src/commands/hello.rs
use clap::Args;

#[derive(Args)]
#[command(about = "Greet a user")]
pub struct HelloArgs {
    /// Name to greet
    pub name: String,

    /// Print greeting in uppercase
    #[arg(short, long)]
    pub uppercase: bool,
}

pub fn run(args: HelloArgs) {
    let greeting = format!("Hello, {}!", args.name);
    println!("{}", if args.uppercase { greeting.to_uppercase() } else { greeting });
}
